battenberg/utils.py

```diff
@@ -4,19 +4,30 @@
 import subprocess
 from typing import Optional
 from pygit2 import discover_repository, init_repository, Keypair, Repository
+from battenberg.errors import InvalidRepositoryException
 
 
 logger = logging.getLogger(__name__)
 
 
 def open_repository(path: str) -> Repository:
-    return Repository(discover_repository(path))
+    try:
+        repo_path = discover_repository(path)
+    except Exception as e:
```

Comment:
is there a more specific exception we could catch here? https://github.com/libgit2/pygit2/blob/9d7e512409f29c967d84dcb8df997b5cf67fbf45/src/error.c#L35

Reply:
Unfortunately not seems like they all come through as vanilla `Exception` types, let me add in a comment here to clarify that. That's why I needed the `if` to check the message.